com.unity.render-pipelines.core/Editor/Lighting/SRPLensFlareElementEditor.cs

```diff
@@ -0,0 +1,490 @@
+                            EditorGUI.PropertyField(rect, colorGradientProp, Styles.colors);
+                            rect = GetNextRect();
+                            EditorGUI.PropertyField(rect, positionCurveProp, Styles.positionCurve);
+                            rect = GetNextRect();
+                            EditorGUI.PropertyField(rect, scaleCurveProp, Styles.scaleCurve);
+                        }
+                    }
+                }
+
+                isFoldOpenedProp.boolValue = true;
+            }
+            else
+            {
+                rect = GetNextRect(35.0f);
+                SRPLensFlareType newType;
+                SRPLensFlareType typeValue = (UnityEngine.SRPLensFlareType)flareTypeProp.enumValueIndex;
+                if ((newType = ((SRPLensFlareType)(EditorGUI.EnumPopup(rect, Styles.flareType, typeValue)))) != typeValue)
+                    flareTypeProp.enumValueIndex = (int)newType;
+                rect = GetNextRect();
+                if ((tmpCol = EditorGUI.ColorField(rect, Styles.tint, tintProp.colorValue)) != tintProp.colorValue)
+                    tintProp.colorValue = tmpCol;
+
+                rect = GetNextRect();
+                if ((tmp = EditorGUI.FloatField(rect, Styles.intensity, intensityProp.floatValue)) != intensityProp.floatValue)
+                    intensityProp.floatValue = Mathf.Max(tmp, 0.0f);
+
+
+                if (allowMultipleElementProp.boolValue)
+                {
+                    rect = GetNextRect();
+                    if ((iTmp = EditorGUI.IntField(rect, Styles.count, countProp.intValue)) != countProp.intValue)
+                        countProp.intValue = Mathf.Max(iTmp, 2);
+                }
+
+                isFoldOpenedProp.boolValue = false;
+            }
+            EditorGUI.EndFoldoutHeaderGroup();
+            EditorGUI.EndProperty();
+        }
+
+        /// <summary>
+        /// Override this method to specify how tall the GUI for this field is in pixels
+        /// </summary>
+        /// <param name="property">The SerializedProperty to make the custom GUI for.</param>
+        /// <param name="label">The label of this property.</param>
+        /// <returns>The height in pixels.</returns>
+        public override float GetPropertyHeight(SerializedProperty property, GUIContent label)
+        {
+            SerializedProperty isFoldOpened = property.FindPropertyRelative("isFoldOpened");
+            SerializedProperty preserveAspectRatio = property.FindPropertyRelative("preserveAspectRatio");
+            SerializedProperty distributionProp = property.FindPropertyRelative("distribution");
+            SerializedProperty flareTypeProp = property.FindPropertyRelative("flareType");
+            SerializedProperty enableDistortionProp = property.FindPropertyRelative("enableRadialDistortion");
+            SerializedProperty allowMultipleElementProp = property.FindPropertyRelative("allowMultipleElement");
```

Comment:
Same here, this should be cached, not searched again at each GetPropertyHeight()

Reply:
Oh right 😞 Sorry I was thinking it is an Editor instead a PropertyDrawer. Yes you have no choice here :/